common/lib/common-utils/src/config.ts

```diff
@@ -0,0 +1,78 @@
+import { Lazy } from "./lazy";
+
+const isLocalStorageAvailable = new Lazy(()=>{
+    const localStorageTestKey = "FluidCommonUtilsLocalStorageTestKey";
+    try {
+        localStorage.setItem(localStorageTestKey, "v");
+        localStorage.removeItem(localStorageTestKey);
+        return true;
+    } catch (e) {
+        return false;
+    }
+});
+
+export function createLocalStorageConfigurationProxy<C extends Record<string, unknown>>(
+    namspace: string, target: C): Readonly<C> {
+    if (isLocalStorageAvailable.value) {
+        return new Proxy<C>(
+            target,
+            new LocalStorageConfigurationProxyHandler<C>(namspace));
+    }
+    return target;
+}
+
+export class LocalStorageConfigurationProxyHandler<T extends Record<string, unknown>> implements ProxyHandler<T> {
+    private readonly configCache = new Map<string, unknown>();
+    constructor(private readonly namespace: string) {
+        if (!isLocalStorageAvailable.value) {
+            throw new Error("Local Storage Not Available.");
+        }
+    }
+
+    public get(target: T, p: string): any {
```

Comment:
what about set()? If I overwrite whole property/object, we should reflect that in this.configCache, right?

Reply:
My thought was config will be immutable, as it gets complex it config change underneath